galaxy/api/internal/search.py

```diff
@@ -247,24 +248,25 @@ def _base_queryset(self):
 
         tags = self.filters.get('tags')
         if tags:
-            tags_qs = models.Tag.objects.filter(name__in=tags)
-            qs = qs.distinct().filter(tags__in=tags_qs)
+            tags_qs = models.Content.objects.only('pk').filter(
+                tags__name__in=tags)
+            qs = qs.filter(pk__in=tags_qs)
 
         deprecated = self.filters.get('deprecated')
         if deprecated is not None:
             qs = qs.filter(repository__deprecated=deprecated)
 
         platforms = self.filters.get('platforms')
         if platforms:
-            platforms_qs = models.Platform.objects.filter(
-                name__in=platforms)
-            qs = qs.distinct().filter(platforms__in=platforms_qs)
+            platforms_qs = models.Content.objects.only('pk').filter(
+                platforms__name__in=platforms)
+            qs = qs.filter(pk__in=platforms_qs)
 
         clouds = self.filters.get('cloud_platforms')
         if clouds:
-            clouds_qs = models.CloudPlatform.objects.filter(
-                name__in=clouds)
-            qs = qs.distinct().filter(cloud_platforms__in=clouds_qs)
+            cloud_qs = models.Content.objects.only('pk').filter(
+                cloud_platforms__name__in=platforms)
```

Comment:
Travis tests fail on this, but guessing this should be 'cloud_platforms__name__in=clouds'

Reply:
Thanks. Fixed now.